### PR TITLE
Fix issues with missing persistence for trade state

### PR DESCRIPTION
**Pull request by chimp1984**
There are some bug reports which suggest that the persistence at shutdown is not working reliable (hard kill, crash,...).
We reduce the persistence interval radically to ensure important data is written immediately (200 ms delay).
Further the `requestPersistence` has been missing in most trade tasks. I guess that happened when merging the large refactoring projects (trade protocol, persistence). If shutdown routine gets executed it did not had any effect as we wrote then the trade state, but if that was missing we might miss important trade state/data changes.
 
Fixes https://github.com/bisq-network/bisq/issues/4806
Fixes https://github.com/bisq-network/bisq/issues/4762

### CODE REVIEW

**Comment by ghost** [comment]
This would not resolve #4806 since `requestPersistence` for PendingTrades is only called twice: when the trade is new, and when the user closes the trade.  There was > 12 hours in between where the trade progressed through all its states with no call to `requestPersistence`.  

**[EDIT]**  Ok, I see you're trying to persist in SIGTERM.  Maybe it would be ok.  If the delay is only 200ms maybe that is enough time before the process is shut down.  It wouldn't help in the case of power cut though.

**Comment by chimp1984** [comment]
Oh, yes you are right. I though at trade task complete we call it but I fear that got lost in some merge. It was a bit complicate with several larger PRs (refactoring of trade protocol + refactoring of persistence).
I will check how to fix it best...

**Comment by chimp1984** [comment]
I added the `requestPersistence` at each data change in trade. Hope I did not miss any. We could add some redundancy if we call it at the `complete()` method of each tradeTask but not sure if thats needed.

**Comment by chimp1984** [comment]
Failing test is resolved by @stejbac (https://github.com/bisq-network/bisq/pull/4789#issuecomment-729586729). Will merge in once hie PR is merged.

**Comment by ghost** [comment]
ACK, tested, fixes #4806.

**Comment by chimp1984** [comment]
I double checked all trade (and sub models) related fields as well as all trade protocol tasks and protocols. Hope I did not miss a call but as there is quite a lot of redundancy I think its safe now.

**Comment by sqrrm** [comment]
I feel it might be cleaner to do the persist in the `complete()` method at the end of each trade task. It feels like that's what's being done with this PR anyway, but separately in each task with the chance of missing it somewhere.

Apart from that I think this looks ok and could be merged.

File: core/src/main/java/bisq/core/trade/protocol/BuyerAsTakerProtocol.java
Comment (ghost) [inline core/src/main/java/bisq/core/trade/protocol/BuyerAsTakerProtocol.java]: I'm getting a NPE here, processModel.tradeManager is null

File: core/src/main/java/bisq/core/trade/protocol/SellerAsTakerProtocol.java
Comment (ghost) [inline core/src/main/java/bisq/core/trade/protocol/SellerAsTakerProtocol.java]: I get a NPE here, processModel.tradeManager is null.

**Review by ghost — CHANGES_REQUESTED** [review]

File: core/src/main/java/bisq/core/trade/protocol/BuyerAsTakerProtocol.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/trade/protocol/BuyerAsTakerProtocol.java]: Ah in the constructor it must not be called as trademanager is not set yet. I will remove that.

File: core/src/main/java/bisq/core/trade/protocol/SellerAsTakerProtocol.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/trade/protocol/SellerAsTakerProtocol.java]: Ah in the constructor it must not be called as trademanager is not set yet. I will remove that.

**Comment by chimp1984** [comment]
> I feel it might be cleaner to do the persist in the `complete()` method at the end of each trade task. It feels like that's what's being done with this PR anyway, but separately in each task with the chance of missing it somewhere.
> 
> Apart from that I think this looks ok and could be merged.

I was wondering myself if we should add at the tradeTask complete handler a perist call. It would add extra redundancy in case I missed one as well in case we add some tasks or have some changes and the call is forgotten. 
I have no strong opinion for or against it. Signal with up or downvote if you want to have it in.

**Comment by sqrrm** [comment]
With the persistence call in complete() it's not necessary to persist in each trade task.

**Comment by chimp1984** [comment]
> With the persistence call in complete() it's not necessary to persist in each trade task.

A reason why I did not do want to rely only on complete is as there are async tasks and i want to be sure that if the task never completes (shutdown) that the intermediate state change is persisted. But also not happy that its that verbose now ;-(. But its more similar as it was before, just that some calls have been done inside the trade setter methods.

**Comment by sqrrm** [comment]
Agreed that some persist calls should still be done in the tasks, but all of those done synchronously just before complete() are no longer needed.

**Pull request merged**
